packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2305,13 +2307,23 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       // export default interface allowed in:
       // https://github.com/Microsoft/TypeScript/pull/16040
       if (this.state.value === "interface") {
-        const result = this.tsParseDeclaration(
-          this.startNode(),
-          this.state.value,
-          true,
-        );
+        const interfaceNode = this.startNode();
+        const prevValue = this.state.value;
+        this.next();
 
-        if (result) return result;
+        if (this.state.type === tt.name) {
+          const result = this.tsParseDeclaration(
+            interfaceNode,
+            prevValue,
```

Comment:
```suggestion
            "interface",
```

`prevValue` is always `"interface"` here (it's guarded in the `if` at line 2309), so we don't need to store it in a variable.